wandb/sdk/wandb_run.py

```diff
@@ -3069,6 +3072,96 @@ def _prepare_artifact(
+            model_name: (str, optional) An artifact name. String containing only the following alphanumeric characters: dashes, underscores, and dots.
+                This will default to the basename of the path prepended with the current
+                run id  if not specified.
+            aliases: (list, optional) Aliases to apply to this artifact,
+                    defaults to `["latest"]`
+
+        Returns:
+            None
+        """
+        self._log_artifact(
+            artifact_or_path=path, name=model_name, type="model", aliases=aliases
+        )
+
+    @_run_decorator._attach
+    def use_model(self, model_name: str) -> FilePathStr:
+        """Download a logged model artifact.
+
+        Arguments:
+            model_name: (str) A model artifact name.
+                May be prefixed with entity/project/. Valid names
+                can be in the following forms:
+                    - name:version
+                    - name:alias
+                    - digest.
+
+        Returns:
+            path: (StrPath) path to downloaded artifact file(s).
+        """
+        r = self._run_obj
+        assert r is not None
+        artifact = self.use_artifact(artifact_or_name=model_name)
+        assert "model" in str(artifact.type.lower())
+        path = artifact.download()
+
+        # If returned directory contains only one file, return path to that file
+        dir_list = os.listdir(path)
+        if len(dir_list) == 1:
+            return FilePathStr(os.path.join(path, dir_list[0]))
+        return path
+
+    def link_model(
+        self,
+        path: StrPath,
+        linked_model_name: str,
+        model_name: Optional[str] = None,
+        aliases: Optional[List[str]] = None,
+    ) -> None:
+        """Link a model version to a model portfolio (a promoted collection of model artifacts).
+
+        The linked model will be visible in the UI for the specified portfolio.
+
+        Arguments:
+            path: (str) A path to the contents of this model,
+                can be in the following forms:
+                    - `/local/directory`
+                    - `/local/directory/file.txt`
+                    - `s3://bucket/path`
+            linked_model_name: (str) - the name of the registered model that the model is to be linked to. The entity will be derived from the run
+            model_name: (str) - the name of the model artifact that files in 'path' will be logged to.
+            aliases: (List[str], optional) - alias(es) that will only be applied on this linked artifact
+                inside the registered model.
+                The alias "latest" will always be applied to the latest version of an artifact that is linked.
+
+        Returns:
+            None
+        """
+        name_parts = linked_model_name.split("/")
+        assert len(name_parts) == 1
+        project = "model-registry"
+        target_path = self.entity + "/" + project + "/" + linked_model_name
```

Comment:
what if we're in offline mode and entity is therefore ''? some actionable check might be quite nice.

Reply:
Wouldn't `run.link_artifact()` take care of that? Although it seems like there is no offline-model implementation for link artifact 😅

Reply:
@dmitryduev Actually, I think i don't understand what actionable check we would add here. I tried running this in offline mode, and the `entity` didn't get set to `''`. 

Even if it did, the path that gets passed to `link_artifact()` would be `/model-registry/<linked_model_name>`, which doesn't cause any issues as the interface _publish method assigns entity from the settings too. Am i missing something?